data/src/main/java/com/teamdontbe/data/repositoryimpl/NotificationRepositoryImpl.kt

```diff
@@ -0,0 +1,34 @@
+package com.teamdontbe.data.repositoryimpl
+
+import com.teamdontbe.data.datasource.NotificationDataSource
+import com.teamdontbe.domain.entity.NotiEntity
+import com.teamdontbe.domain.repository.NotificationRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class NotificationRepositoryImpl
+    @Inject
+    constructor(
+        private val notificationDataSource: NotificationDataSource,
+    ) : NotificationRepository {
+        override suspend fun getNotificationList(): Flow<List<NotiEntity>?> {
+            return flow {
+                val result =
+                    runCatching {
+                        notificationDataSource.getNotificationList().data?.map { it.toNotificationEntity() }
+                    }
+                emit(result.getOrDefault(emptyList()))
+            }
+        }
+
+        override suspend fun getNotificationCount(): Flow<Int?> {
+            return flow {
+                val result =
+                    runCatching {
+                        notificationDataSource.getNotificationCount().data?.notificationNumber
+                    }
+                emit(result.getOrDefault(0))
```

Comment:
```suggestion
                emit(result.getOrDefault(-1))
```
p1 : 기본값 -1로 해주세융 절대로 받을 수 없는 값!

Reply:
수정 및 예외 처리 로직 추가 완~